Fix moment 0 calculation for spectral axis that isn't last, update spatial moment

rosteen opened the pull request:
Non-spectral axis moments now use pixel values, previously they incorrectly still used the spectral axis for `dx` and `dispersion` and then simply did the final sums over the specified axis. It still may be reasonable to just get rid of non-spectral moments as discussed in https://github.com/astropy/specutils/issues/930.

keflavich reviewed (commented):
lgtm, I just have implementation nitpicks.

I think you need to add the expected units to the docstring, though.

rosteen commented on specutils/analysis/moment.py:
```diff
@@ -51,6 +51,9 @@ def _compute_moment(spectrum, regions=None, order=0, axis='spectral'):
     """
     This is a helper function for the above `moment()` method.
     """
+    if order is None or order < 0:
+        return None
+
```
Good call - this is just existing code moved from further down but might as well improve it while I'm at it.

rosteen commented on specutils/analysis/moment.py:
```diff
@@ -71,24 +74,30 @@ def _compute_moment(spectrum, regions=None, order=0, axis='spectral'):
+    # If axis is not the spectral axis, we simply use pixel values for dx and dispersion
+    if (axis != spectrum.spectral_axis_index and
+            not (axis == -1 and spectrum.spectral_axis_index == spectrum.flux.ndim-1)):
+        dx = np.ones(flux.shape[axis])
```
I think it's worth doing it this way to not have to handle the two cases differently below.

rosteen commented:
> lgtm, I just have implementation nitpicks.
> 
> I think you need to add the expected units to the docstring, though.

Alright, I addressed your comments, thanks for the further thoughts!

rosteen commented:
Test failures are known and unrelated.
